ui/app/AppLayouts/Onboarding2/KeycardFactoryResetFlow.qml

```diff
@@ -0,0 +1,108 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1 as SQUtils
+
+import AppLayouts.Onboarding2.pages 1.0
+import AppLayouts.Onboarding.enums 1.0
+
+SQUtils.QObject {
+    id: root
+
+    required property StackView stackView
+    required property int keycardState
+
+    signal performKeycardFactoryResetRequested
+
+    signal finished
+
+    function init(fromLoginScreen = false) {
+        d.fromLoginScreen = fromLoginScreen
+        root.stackView.push(d.initialComponent())
+    }
+
+    QtObject {
+        id: d
+
+        property bool fromLoginScreen
+
+        function initialComponent() {
+            if (root.keycardState === Onboarding.KeycardState.FactoryResetting)
+                return keycardResetPageComponent
+            return keycardResetAcks
+        }
+    }
+
+    Component {
+        id: keycardResetAcks
+
+        KeycardBasePage {
+            image.source: Theme.png("onboarding/keycard/reading")
+            title: qsTr("Factory reset Keycard")
+            subtitle: "<font color='%1'>".arg(Theme.palette.dangerColor1) +
```

Comment:
Why not expose `subtitleColor` property? 🤔

Reply:
Hmm, good question :) I can expose it but this way works as well